x-pack/plugins/security_solution/public/management/pages/trusted_apps/view/components/condition_entry_input/index.test.tsx

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { shallow, mount } from 'enzyme';
+import React from 'react';
+import {
+  ConditionEntry,
+  ConditionEntryField,
+  OperatingSystem,
+} from '../../../../../../../common/endpoint/types';
+
+import { ConditionEntryInput } from '.';
+import { EuiSuperSelectProps } from '@elastic/eui/src/components/form/super_select/super_select';
+
+let onRemoveMock: jest.Mock;
+let onChangeMock: jest.Mock;
+let onVisitedMock: jest.Mock;
+
+const entry: ConditionEntry = {
+  field: ConditionEntryField.HASH,
+  type: 'match',
+  operator: 'included',
+  value: 'trustedApp',
+};
+
+describe('Condition entry input', () => {
+  beforeEach(() => {
+    onRemoveMock = jest.fn();
+    onChangeMock = jest.fn();
+    onVisitedMock = jest.fn();
+  });
+
+  const getElement = (
+    subject: string,
+    os: OperatingSystem = OperatingSystem.WINDOWS,
+    isRemoveDisabled: boolean = false
+  ) => (
+    <ConditionEntryInput
+      os={os}
+      entry={entry}
+      showLabels
+      onRemove={onRemoveMock}
+      onChange={onChangeMock}
+      onVisited={onVisitedMock}
+      data-test-subj={subject}
+      isRemoveDisabled={isRemoveDisabled}
+    />
+  );
+
+  for (const field in ConditionEntryField) {
+    if (Object.prototype.hasOwnProperty.call(ConditionEntryField, field)) {
```

Comment:
curious - why did you need this check here? was the for loop returning `prototype` properties?

Reply:
even if not, I think this is always _technically_ best practice JS to do it this way.

Reply:
It was because an eslint check:
``` 
 55:3   error  The body of a for-in should be wrapped in an if statement to filter unwanted properties from the prototype  guard-for-in
 ```